cmd/harmony/main.go

```diff
@@ -131,6 +135,20 @@ func prepareRootCmd(cmd *cobra.Command) {
 	runtime.GOMAXPROCS(runtime.NumCPU())
 	// Set up randomization seed.
 	rand.Seed(int64(time.Now().Nanosecond()))
+	// Raise fd limits
+	return raiseFdLimits()
+}
+
+func raiseFdLimits() error {
```

Comment:
does this require the user to run ./harmony as root?

Reply:
how is .Maximum() determined?

Reply:
No, not necessarily.

Reply:
By system configuration cap and a hard cap defined in the module. This piece of code is copied from ethereum code.